File: JUNHEE/프로그래머스_week6/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>

using namespace std;
int measure (int num){
    int count = 0;
    for(int i = 1; i * i < num + 1; i++){
        if(num % i == 0){
            if(i*i != num) count += 2;
            else count += 1;
        }
    }
    return count;
}
int solution(int left, int right) {
    int answer = 0;
    for(int i = left; i < right + 1; i++){
        if(measure(i) % 2 == 0) answer += i;
        else answer -= i;
    }
    return answer;
}

File: DONGUK/프로그래머스_WEEK 6/약수의 개수와 덧셈.py
def count(num):
    cnt = 0
    for i in range(1, num+1):
        if num % i == 0:
            cnt +=1
    
    return cnt

def solution(left, right):
    answer = 0
    
    for num in range(left, right+1):
        ret = count(num)
        if ret % 2 == 0:
            answer += num
            
        else:
            answer -= num
    
    return answer